### DISCUSSION TITLE
Unicode 14.0 + Emoji 14.0 in Nerd Fonts

### DISCUSSION THREAD

**Discussion by mkanet** [ryanoasis/nerd-fonts#1145] [Q&A]
I noticed that the Unicode character: ["Identification Card"](https://emojipedia.org/identification-card/) is not available in the latest Sauce Code Pro downloadable on the Nerd Fonts website.

It looks like that Unicode character is part of Unicode 14.0 + Emoji 14.0 character sets.  I'm curious when will we see this Unicode character in Sauce Code Pro Nerd Font?  I didn't want to make a custom patch if this was already in the near-future roadmap.

Thank you for all your efforts maintaining Nerd Fonts.

**Comment by Finii** [comment]
At the moment the Emoji are not patched in at all (i.e. codepoints `1Fxxx`).
Patching those in is not on any roadmap.

Usually people have the Emoji as color glyphs, for example via Noto Color Emoji and Font-Fallback.
I guess very precise positioning is not needed for those, at least when they are used as emoji; so font fallback is a good method.
This is different if you want to have it in your prompt of course.

Another issue is that patching a color font is ... difficult to impossible with `fontforge`. That would mean to patch them as black-and-white, and I guess a lot of people would not like that. I did think about patching with `Glyphs`, which might work for the prepatched fonts, but that would mean no self patching and/or parallel maintenance of 2 different methods, which I guess is impossible.

But maybe you can use one of the many similar looking glyphs instead:

![image](https://user-images.githubusercontent.com/16012374/224472468-e52316ba-cd1e-44b6-84ae-195866266d44.png)

**Reply by mkanet** [reply]
Thank you for taking the time to explain.  I'm embarrassed to say I didn't fully understand since I'm very new to using Nerd Fonts. 

Is it possible for me to use a specific color Emoji from `Noto Color Emoji` via Font-Fallback for my prompt?  Or, is this method only meant for CSS/HTML?

If it's not possible, no worries, I can continue to pick from the included characters in Sauce Code Pro Complete Nerd Font.

**EDIT:**  Is there an easy way for me to see/print all **available colorized characters** in Sauce Code Pro?  The cheat-sheet above seems to only display non-colorized font characters.  Currently I'm just searching for unicode character codes online one-by-one...  and printing each unicode character to see see if it is available in color in Sauce Code Pro or not.

**Reply by Finii** [reply]
To answer the last question first _How to see all colorized glyphs in Sauce Code Pro NF_. That is simple: It's a standard black and white font, there are no color glyphs.

I never really looked into the color glyphs, but this is what `Noto Color Emoji` says:

![image](https://user-images.githubusercontent.com/16012374/224555428-01514408-5586-4408-ba5a-62ba74cbf8da.png)
_https://notofonts.github.io/noto-docs/specimen/NotoColorEmoji/_

It is not recognized by fontconf on my system (Ubuntu), so font fallback is not working. It IS detected by Firefox iirc. But that does not help with your problem.

Then there is the same font in monochrome:

![image](https://user-images.githubusercontent.com/16012374/224555586-3e5d8182-4462-46dc-9741-e9bc24e94020.png)
_https://notofonts.github.io/noto-docs/specimen/NotoEmoji/_

And this works in my terminal without any hazzle / special things:

![image](https://user-images.githubusercontent.com/16012374/224555910-c2e4455c-398e-4ba1-aa5e-db3a3c3a1684.png)

But iirc you are on Windows? I do not really have ever tested fontfallback on Windows. There is [this](https://github.com/ryanoasis/nerd-fonts/wiki/Font-Substitution-&-Fallback#windows) entry in the wiki. Which might work or not, esp in Windows Terminal that has special font treatment. You maybe need to ask [there](https://github.com/microsoft/terminal).

**Reply by Finii** [reply]
Just out of curiosity checked mac, seems to supported out of the box, and `iTerm2` even seems to support color glyphs 🎉 

<img width="440" alt="Screenshot 2023-03-12 at 16 58 31" src="https://user-images.githubusercontent.com/16012374/224556943-a6118024-5382-4945-a9be-ff6a2d67595f.png">

I'm no Mac guy, so I have no clue how that works.

P.S. Standard Mac terminal works the same.

**Reply by mkanet** [reply]
Just to be clear, I don't have any issues displaying color glyphs/emojis on Windows.  As far as I know, it has always worked under Win10/Win11 (see below).  However, I incorrectly thought that these emojis were part of Nerd Fonts.  They are actually built into Windows 10/Windows 11 natively.

**Win11**
![Color-Glyphs-Win11](https://user-images.githubusercontent.com/811008/224580457-4688394a-85eb-4e33-b470-168466e4db55.png)

That's how I'm able to display them in my prompt on Windows 10:
![Color-Glyphs-Win10](https://user-images.githubusercontent.com/811008/224580470-13054d24-52f7-4efd-99e8-84279d5903f6.png)

**However, ultimately, my issue is that `\u1faaa` is not available on Windows 10:**  
![Win10](https://user-images.githubusercontent.com/811008/224581020-54299fca-def6-4067-ab71-f6ce1ebbbcec.png)

Since I can't add the missing emoticon to Sauce Code Pro Nerd Font without ruining the color, I am looking for another way to add it to Windows 10, if possible until I upgrade to Win11.
